ibis/backends/duckdb/__init__.py

```diff
@@ -170,6 +171,39 @@ def register(
         self.con.execute(view)
         return self.table(table_name)
 
+    def to_pyarrow_batches(
+        self,
+        expr: ir.Expr,
+        params: Mapping[ir.Scalar, Any] | None = None,
+        limit: int | str | None = None,
+        chunk_size: int = 1_000_000,
+    ) -> Iterator[pa.RecordBatch]:
+        self._check_pyarrow_install()
+        query_ast = self.compiler.to_ast_ensure_limit(
+            expr, limit, params=params
+        )
+        sql = query_ast.compile()
+
+        with self._safe_raw_sql(sql) as cursor:
+            yield from cursor.cursor.fetch_record_batch(chunk_size=chunk_size)
```

Comment:
Just so I understand this code correctly, it's calling down into the `ConnectionWrapper` instance in `duckdb-engine`,  through `__getattr__` on `ConnectionWrapper`?

Reply:
Sorry for the delay -- yes, that's what it's doing.